vignettes/Workflow.Rmd

```diff
@@ -0,0 +1,494 @@
+```
+
+```{r fig=TRUE, height=5, echo=FALSE}
+plot(trellis.last.object())
+```
+
+# Differential peaks
+
+One common question is: which peaks are different in two samples?  One
+simple strategy is the following: combine the two peak sets, and
+compare the two samples by calculating summary statistics for
+the combined peaks on top of each coverage vector.
+
+```{r}
+## find peaks for GFP control
+cov.gfp <- coverage(resize(cstest$gfp, 200))
+peaks.gfp <- slice(cov.gfp, lower = 8)
+
+peakSummary <- diffPeakSummary(peaks.gfp, peaks.ctcf)
+
+xyplot(asinh(sums2) ~ asinh(sums1) | seqnames,
+       data = as.data.frame(peakSummary), 
+       panel = function(x, y, ...) {
+           panel.xyplot(x, y, ...)
+           panel.abline(median(y - x), 1)
+       },
+       type = c("p", "g"), alpha = 0.5, aspect = "iso")
+``` 
+
+```{r fig=TRUE, height=5, echo=FALSE}
+plot(trellis.last.object()) 
+```
+
+We use a simple cutoff to flag peaks that are different. 
+
+```{r}
+mcols(peakSummary) <- 
+    within(mcols(peakSummary), {
+           diffs <- asinh(sums2) - asinh(sums1) 
+           resids <- (diffs - median(diffs)) / mad(diffs) 
+           up <- resids > 2 
+           down <- resids < -2 
+           change <- ifelse(up, "up", ifelse(down, "down", "flat")) 
+    })
+```
+
+# Placing peaks in genomic context 
+
+Locations of individual peaks may be of interest. Alternatively, a
+global summary might look at classifying the peaks of interest in the
+context of genomic features such as promoters, upstream regions, etc.
+The `r Biocpkg("GenomicFeatures")` package facilitates obtaining gene
+annotations from different data sources. We could download the UCSC gene
+predictions for the mouse genome and generate a *GRanges* object with
+the transcript regions (from the first to last exon, contiguous) using
+`makeTxDbFromUCSC`; here we use a library containing a recent snapshot.
+
+```{r}
+library(TxDb.Mmusculus.UCSC.mm9.knownGene) 
+gregions <- transcripts(TxDb.Mmusculus.UCSC.mm9.knownGene) 
+gregions 
+```
+
+We can now estimate the promoter for each transcript: 
+
+```{r}
+promoters <- flank(gregions, 1000, both = TRUE) 
+```
+
+And count the peaks
+that fall into a promoter:
+
+```{r}
+peakSummary$inPromoter <- peakSummary %over% promoters
+xtabs(~ inPromoter + change, peakSummary) 
+```
+
+or somewhere upstream or in a gene:
+
+```{r}
+peakSummary$inUpstream <- peakSummary %over% flank(gregions, 20000)
+peakSummary$inGene <- peakSummary 
+```
+
+```{r}
+sumtab <- 
+   as.data.frame(rbind(total = xtabs(change, peakSummary), 
+                       promoter = xtabs(change, 
+                         subset(peakSummary, inPromoter)), 
+                       upstream = xtabs(change, 
+                         subset(peakSummary, inUpstream)), 
+                       gene = xtabs(change, subset(peakSummary, inGene))))
+## cbind(sumtab, ratio = round(sumtab[, "down"] / sumtab[, "up"], 3))
+```
+
```

Comment:
This loop is looking for the variable "_change_".  "_change_" is populated in another loop on line 389.  

The variableis local to the loop. The variables don't exist outside the loop.

Reply:
Hi @villafup Based on the error message  @mcarlsn outlined, it seems like the issue is with the subset function being used in combination with xtabs. The subset function is not being evaluated correctly, resulting in the error. Specifically, the error message mentions that the 'subset' argument must be coercible to logical, meaning it should be a valid logical expression or a logical vector.

To fix this issue, I want to review and correct the usage of the 'subset', 'as.data.frame', and 'xtabs' functions in the code, making sure that the 'subset' argument is a valid logical expression or vector. 

If you try running my converted code chunk you'd get the error messagge "Error in h(simpleError(msg, call)) : 
  error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': object 'change' not found
Calls: <Anonymous> ... makeEnvForNames -> new.env -> environment -> " and if you try running the exact code chunk from the .Rwn sumtab <- 
    as.data.frame(rbind(total = xtabs(~ change, peakSummary),
                        promoter = xtabs(~ change, 
                          subset(peakSummary, inPromoter)),
                        upstream = xtabs(~ change, 
                          subset(peakSummary, inUpstream)),
                        gene = xtabs(~ change, subset(peakSummary, inGene))))
##cbind(sumtab, ratio = round(sumtab[, "down"] /  sumtab[, "up"], 3)) you'll get a different but similar error message "error in evaluating the argument 'x' in selecting a method for function 'as.data.frame': error in evaluating the argument 'data' in selecting a method for function 'xtabs': 'subset' must be coercible to logical
Calls: <Anonymous> ... subset -> .local -> evalqForSubset -> normSubsetIndex" . What do you advise please?

Reply:
Hi, sorry it took me awhile to look at this issue. I noticed that there are some formatting issues in the code blocks, which can be contributing to this issue. I didn't find exactly where the problem exists, but I tried copying over all code blocks from the .Rnw to the .Rmd and was able to knit and run `R CMD build`. 

From that, I made a patch (but renamed the extension from .patch to .txt so I could share it) so you can see the differences. If you've never looked at a patch or a diff, they show the differences between the files. I recommend looking it with something that has syntax highlighting.

[0001-Copy-code-blocks-from-.Rnw.txt](https://github.com/Bioconductor/chipseq/files/11160186/0001-Copy-code-blocks-from-.Rnw.txt)

You could also copy the code blocks from the .Rnw then run knit incrementally to identify the problem.  If you understand the content of the patch, you could also rename and it.

Reply:
Hello @jwokaty Hope your day is going great? So I was able to knit and run R CMD build after copying over all code blocks from the .Rnw to the .Rmd like you advised. Now, I'm trying to compare both code from both files to find the cause of the issue in the .Rmd file. Hopeful I figure out soon and send a new commit. Thank you!